src/main.rs

```diff
@@ -0,0 +1,188 @@
+use std::error::Error;
+use std::ffi::{CStr, CString};
+use std::fmt::{Display, Formatter};
+use winit::event_loop::EventLoop;
+use winit::raw_window_handle::HasWindowHandle;
+use winit::window::{Window, WindowAttributes};
+
+use glutin::config::{Config, ConfigTemplateBuilder};
+use glutin::context::{ContextApi, ContextAttributesBuilder, NotCurrentContext, Version};
+use glutin::display::GetGlDisplay;
+use glutin::prelude::*;
+
+use glutin_winit::{DisplayBuilder, GlWindow};
+
+pub mod gl {
+    #![allow(clippy::all)]
```

Comment:
```suggestion
    #![allow(clippy::all)]
    #![allow(unsafe_op_in_unsafe_fn)]
```
https://doc.rust-lang.org/stable/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html

Reply:
Awesome, thanks!